src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/files/EjbClient.java.src

```diff
@@ -28,7 +28,7 @@ import jakarta.ejb.SessionContext;
 import jakarta.ejb.TransactionManagement;
 import jakarta.ejb.TransactionManagementType;
 import jakarta.jms.*;
-import javax.transaction.*;
+import jakarta.transaction.*;
```

Comment:
I could be overlooking something but I don't see any javax.transaction or jakarta.transaction classes referenced in EjbClient.jsp.

Reply:
Maybe we can get a test run with the rename before we try deleting code.  I don't fully understand the `*.src` files am a bit nervous about having two potential reasons a test failed; i.e. was it the rename or the delete.

Reply:
Currently we are only including Java SE 8 javax.transaction classes with the ```javax.transaction.*```:
```
javax.transaction.InvalidTransactionException
javax.transaction.TransactionRequiredException
javax.transaction.TransactionRolledbackException
javax.transaction.xa.XAException
javax.transaction.xa.XAResource
javax.transaction.xa.Xid
```

With the fix, we will instead include ```jakarta.transaction.*``` classes.  So, the lower risk thing would be to keep the import of javax.transaction.* to continue importing the Java SE transaction classes.

I'm not sure about these .src files either, they look like ignored files but not 100% sure.

Do we have any Platform TCK build jobs that can optionally build from a set of pull requests (or even just one pull request) without hacking the Jenkins job each time?  It would help if the pull requests could be specified via a Jenkins job parameter.

Reply:
Note that these classes do not exist in Java SE anymore, have been renamed to `jakarta` and if any part of the TCK is using them it is a show stopper:

 - javax.transaction.InvalidTransactionException
 - javax.transaction.TransactionRequiredException
 - javax.transaction.TransactionRolledbackException

These are legitimate Java SE classes, absolutely correct to use as `javax` and should not be referred to as `jakarta`

 - `javax.transaction.xa.XAException`
 - `javax.transaction.xa.XAResource`
 - `javax.transaction.xa.Xid`

We can add an import to `javax.transaction.xa.*`, but there's no way for us to release a final TCK with references to `javax.transaction.*` even if the file is assumed to be not needed.

We'd need to either delete the import, change the import or delete the file.  I'm ok with any of those solutions.

Reply:
> Note that these classes do not exist in Java SE anymore, have been renamed to jakarta and if any part of the TCK is using them it is a show stopper:
> 
>     javax.transaction.InvalidTransactionException
>     javax.transaction.TransactionRequiredException
>     javax.transaction.TransactionRolledbackException

I verified that the Platform TCK does not reference { javax.transaction.InvalidTransactionException, javax.transaction.TransactionRequiredException, javax.transaction.TransactionRolledbackException }.

> We can add an import to javax.transaction.xa.*, but there's no way for us to release a final TCK with references to javax.transaction.* even if the file is assumed to be not needed.

I did look for { "xid",  "xa" } in each of the updated files and do not see any references to xid or xa.

> We'd need to either delete the import, change the import or delete the file. I'm ok with any of those solutions.

Deleting the import makes the most sense to me since they aren't likely used.

Reply:
I feel the *.src files in this folder itself could be removed as they looks like backup copy of some files under https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations.

Reply:
I looked a bit deeper and do see at least java.src file is actually being used (ant script does some replacement of raname, username, password with the result copied to a .java file).  So, we should keep them around I think.

More details...

Just to see how many of these backup copy or whatever they are files there are ```find -name *.src```
```
./src/com/sun/ts/tests/appclient/deploy/metadatacomplete/testapp/files/TestBean.java.src
./src/com/sun/ts/tests/appclient/deploy/metadatacomplete/testapp/files/DataStoreBean.java.src
./src/com/sun/ts/tests/appclient/deploy/metadatacomplete/testapp/files/TestAppClient.java.src
./src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/files/ServletClient.java.src
./src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/files/Client.java.src
./src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/files/EjbClient.java.src
./src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/files/MyBean.java.src
./src/com/sun/ts/tests/common/vehicle/wsejb/wsejb_vehicle_client.xml.src
./src/com/sun/ts/tests/common/vehicle/wsejb/WSEJBVehicle.java.src
./src/com/sun/ts/tests/common/vehicle/wsappclient/WSAppclient.java.src
./src/com/sun/ts/tests/common/vehicle/wsservlet/WSServletVehicle.java.src
```

I see to references to "\.src\"" in:
```
src/com/sun/ts/tests/appclient/deploy/metadatacomplete/testapp/build.xml
src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/build.xml
src/com/sun/ts/tests/common/vehicle/wsejb/build.xmlsrc/com/sun/ts/tests/common/vehicle/wsappclient/build.xml
src/com/sun/ts/tests/common/vehicle/wsservlet/build.xml
src/com/sun/ts/tests/jsf/spec/ajax/common/ajax.common.xml
```

From src/com/sun/ts/tests/jms/ee20/resourcedefs/annotations/build.xml:
```
 <target name="do.token.substitution.of.java.src.files">
    <copy failonerror="false" overwrite="true" 
      file="${ts.home}/src/${pkg.dir}/files/Client.java.src"
      tofile="${ts.home}/src/${pkg.dir}/Client.java">
        <filterset>
          <filter token="raname" value="${jmsra.name}"/>
          <filter token="user" value="${user}"/>
          <filter token="password" value="${password}"/>
        </filterset>
    </copy>
    <copy failonerror="false" overwrite="true" 
      file="${ts.home}/src/${pkg.dir}/files/EjbClient.java.src"
      tofile="${ts.home}/src/${pkg.dir}/EjbClient.java">
        <filterset>
          <filter token="raname" value="${jmsra.name}"/>
          <filter token="user" value="${user}"/>
          <filter token="password" value="${password}"/>
        </filterset>
    </copy>
    <copy failonerror="false" overwrite="true" 
      file="${ts.home}/src/${pkg.dir}/files/ServletClient.java.src"
      tofile="${ts.home}/src/${pkg.dir}/ServletClient.java">
        <filterset>
          <filter token="raname" value="${jmsra.name}"/>
          <filter token="user" value="${user}"/>
          <filter token="password" value="${password}"/>
        </filterset>
    </copy>
    <copy failonerror="false" overwrite="true" 
      file="${ts.home}/src/${pkg.dir}/files/MyBean.java.src"
      tofile="${ts.home}/src/${pkg.dir}/MyBean.java">
        <filterset>
          <filter token="raname" value="${jmsra.name}"/>
          <filter token="user" value="${user}"/>
          <filter token="password" value="${password}"/>
        </filterset>
    </copy>
  </target>
```

Reply:
Thanks. Then the options remain are to delete or change the import.